x-pack/plugins/security_solution/public/common/containers/sourcerer/index.tsx

```diff
@@ -88,6 +92,41 @@ export const useInitSourcerer = (
   } = useDeepEqualSelector((state) => scopeIdSelector(state, SourcererScopeName.timeline));
   const { indexFieldsSearch } = useDataView();
 
+  const onInitializeUrlParam = useCallback(
+    (initialState: SourcererUrlState | null) => {
+      // Initialize the store with value from UrlParam.
+      if (initialState != null) {
+        (Object.keys(initialState) as SourcererScopeName[]).forEach((scope) => {
+          if (
+            !(scope === SourcererScopeName.default && scopeId === SourcererScopeName.detections)
+          ) {
+            dispatch(
+              sourcererActions.setSelectedDataView({
+                id: scope,
+                selectedDataViewId: initialState[scope]?.id ?? null,
+                selectedPatterns: initialState[scope]?.selectedPatterns ?? [],
+              })
+            );
+          }
+        });
+      } else {
+        // Initialize the UrlParam with values from the store.
+        // It isn't strictly necessary but I am keeping it for compatibility with the previous implementation.
+        if (scopeDataViewId) {
```

Comment:
Is there a current scenario where this will execute?

Reply:
Yes, when you load a security solution page without sourcerer URL parameter. 
Scenario 1: 
* Open kibana
* On the landing page, click on the security link
* Page loads and updates URL with default selected patterns 

Scenario 1: 
* Given you are on a security solution page
* Update the data view and select only one pattern
*  Navigate to Kibana observability
* Navigate back to security
* The query string should have only one pattern selected